reader/options.go

```diff
@@ -0,0 +1,67 @@
+package reader
+
+import (
+	"github.com/spy16/parens/value"
+)
+
+// Option for Reader
+type Option func(*Reader)
+
+// MacroTable is a lookup table that maps the first rune of a macro expression to its
+// implementation.
+type MacroTable map[rune]Macro
+
+// WithMacros sets the macro table for the reader
+func WithMacros(t MacroTable) Option {
```

Comment:
Issue with this Option is it forces overriding macros an all-or-nothing configuration. What issue did you face with `(*Reader).SetMacro()` method (which allows overriding any macro/dispatch individually)?

Reply:
It was just for convenience.  For overriding single-macros, the `SetMacro` method is fine.  In my case, I was overwriting _all_ the macros since I have custom implementations for each type.

To be clear:  I could have just called `SetMacro` repeatedly.  I don't mind if we remove this option.